packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -0,0 +1,98 @@
+import VerificationReasons from '../../constants/verification-reasons';
+import { BeginSigninError, NavigationOptions } from './interfaces';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../lib/auth-errors/auth-errors';
+
+export const getNavigationTarget = ({
+  email,
+  sessionVerified,
+  verificationReason,
+  verificationMethod,
+  verified,
+  wantsTwoStepAuthentication,
+}: NavigationOptions) => {
+  // Note, all navigations are missing query params. Add these when working on
+  // subsequent tickets.
+  if (!verified) {
+    // TODO: Does force password change ever reach here, or can we move
+    // CHANGE_PASSWORD checks to another page?
+    if (
+      ((verificationReason === VerificationReasons.SIGN_IN ||
+        verificationReason === VerificationReasons.CHANGE_PASSWORD) &&
+        verificationMethod === VerificationMethods.TOTP_2FA) ||
+      wantsTwoStepAuthentication
+    ) {
+      // TODO with signin_totp_code ticket, content server says this (double check it):
+      // Login requests that ask for 2FA but don't have it setup on their account
+      // will return an error.
+      return {
+        to: '/signin_totp_code',
+        options: {
+          state: { verificationReason, verificationMethod },
+        },
+      };
+    } else if (verificationReason === VerificationReasons.SIGN_UP) {
+      // do we need this?
+      // if (verificationMethod !== VerificationMethods.EMAIL_OTP) {
+      //  send email verification since this screen doesn't do it automatically
+      // }
+      return { to: '/confirm_signup_code', options: undefined };
+    } else {
+      // TODO: Pretty sure we want this to be the default. The check used to be:
+      // if (
+      //   verificationMethod === VerificationMethods.EMAIL_OTP &&
+      //   (verificationReason === VerificationReasons.SIGN_IN || verificationReason === VerificationReasons.CHANGE_PASSWORD)) {
+      return {
+        to: '/signin_token_code',
+        options: {
+          state: {
+            email,
+            // TODO: We may want to store this in local apollo cache
+            // instead of passing it via location state, depending on
+            // if we reference it in another spot or two and if we need
+            // some action to happen dependent on it that should occur
+            // without first reaching /signin.
+            verificationReason,
+          },
+        },
+      };
+    }
+    // Verified account, but session hasn't been verified
+  } else if (sessionVerified === false) {
+    return { to: '/signin_token_code', options: undefined };
+  } else {
+    return { to: '/settings', options: undefined };
+  }
+};
+
+export const handleGQLError = (error: any) => {
+  const graphQLError: GraphQLError = error.graphQLErrors?.[0];
+  const errno = graphQLError?.extensions?.errno as number;
+
+  if (errno && AuthUiErrorNos[errno]) {
+    const uiError = {
+      name: AuthUiErrorNos[errno].name,
```

Comment:
`name` was included to match the `AuthUiError` type that was expected by `getLocalizedErrorMessage`, but really not necessary in this instance. 

Good idea to use the `HandledError` type. I'll update the `getLocalizedErrorMessage` to allow errors of either `AuthUiError` or `HandledError` types, which will allow us to simplify the shape of the returned error to only include the essentials - `errno`, `message`, and `retryAfter` values.

Reply:
Wondering if we should file a ticket for a more holistic review of error handling in fxa-settings since we've had a few people working on these in slightly different ways (though we've all tried to be as consistent as we can).